modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/upgrade/test/SXPBlueprintUpgradeProcessTest.java

```diff
@@ -0,0 +1,480 @@
+			sxpBlueprintWithDisableAllContributors);
+		_assertGeneralConfigurationExcludesAndIncludes(
+			new String[0], wildcardArray,
+			sxpBlueprintWithEnableAllContributors);
+
+		String[] expectedIncludes = {
+			"com.liferay.journal.internal.search.spi.model.query.contributor." +
+				"JournalArticleKeywordQueryContributor",
+			"com.liferay.journal.internal.search.spi.model.query.contributor." +
+				"JournalFolderKeywordQueryContributor"
+		};
+
+		_assertGeneralConfigurationExcludesAndIncludes(
+			new String[0], expectedIncludes,
+			sxpBlueprintWithEnableSomeContributors);
+	}
+
+	private PortletPreferenceValue _addPortletPreferenceValue(
+			String name, String portletId, String largeValue, String smallValue)
+		throws Exception {
+
+		PortletPreferences portletPreferences =
+			_portletPreferencesLocalService.addPortletPreferences(
+				TestPropsValues.getCompanyId(), _group.getGroupId(),
+				PortletKeys.PREFS_OWNER_TYPE_LAYOUT, TestPropsValues.getPlid(),
+				portletId + "_INSTANCE_" + RandomTestUtil.randomString(),
+				_portletLocalService.getPortletById(portletId),
+				PortletConstants.DEFAULT_PREFERENCES);
+
+		PortletPreferenceValue portletPreferenceValue =
+			_portletPreferenceValueLocalService.createPortletPreferenceValue(
+				RandomTestUtil.randomLong());
+
+		portletPreferenceValue.setCompanyId(TestPropsValues.getCompanyId());
+		portletPreferenceValue.setPortletPreferencesId(
+			portletPreferences.getPortletPreferencesId());
+		portletPreferenceValue.setIndex(0);
+		portletPreferenceValue.setLargeValue(largeValue);
+		portletPreferenceValue.setName(name);
+		portletPreferenceValue.setReadOnly(false);
+		portletPreferenceValue.setSmallValue(smallValue);
+
+		return _portletPreferenceValueLocalService.addPortletPreferenceValue(
+			portletPreferenceValue);
+	}
+
+	private SXPBlueprint _addSXPBlueprint(
+			String configurationJSON, String elementInstancesJSON,
+			String schemaVersion)
+		throws Exception {
+
+		SXPBlueprint sxpBlueprint = _sxpBlueprintLocalService.addSXPBlueprint(
+			null, _user.getUserId(), configurationJSON,
+			Collections.singletonMap(LocaleUtil.US, ""), elementInstancesJSON,
+			schemaVersion,
+			Collections.singletonMap(
+				LocaleUtil.US, RandomTestUtil.randomString()),
+			_serviceContext);
+
+		_sxpBlueprintIds.add(sxpBlueprint.getSXPBlueprintId());
+
+		return sxpBlueprint;
+	}
+
+	private SXPElement _addSXPElement(String elementDefinitionJSON)
+		throws Exception {
+
+		SXPElement sxpElement = _sxpElementLocalService.addSXPElement(
+			null, _user.getUserId(),
+			Collections.singletonMap(LocaleUtil.US, ""), elementDefinitionJSON,
+			"", "", true, "",
+			Collections.singletonMap(
+				LocaleUtil.US, RandomTestUtil.randomString()),
+			0, _serviceContext);
+
+		_sxpElementIds.add(sxpElement.getSXPElementId());
+
+		return sxpElement;
+	}
+
+	private void _assertGeneralConfigurationExcludesAndIncludes(
+			String[] expectedExcludes, String[] expectedIncludes,
+			SXPBlueprint sxpBlueprint)
```

Comment:
just pass the blueprint id since it is the only necessary information the method needs.